ArduPlane/quadplane.cpp

```diff
@@ -2793,6 +2793,13 @@ void QuadPlane::vtol_position_controller(void)
             loc2.change_alt_frame(Location::AltFrame::ABOVE_ORIGIN);
             float target_z = loc2.alt;
             float zero = 0;
+#if AP_TERRAIN_AVAILABLE
+            float terrain_altitude_offset;
+            if (plane.terrain_enabled_in_current_mode() && plane.terrain.height_terrain_difference_home(terrain_altitude_offset, true)) {
+                // Climb if current terrain is above home, target_altitude_cm is reltive to home
+                target_z += MAX(terrain_altitude_offset*100,0);
```

Comment:
because we could dive down into a canyon, and not be able to make it back up again. Same reason we don't descend in the code a few lines up

Reply:
That is not our standard terrain following behavior. It also does not prevent the case you describe. It allows any climb and descending so long as you remain above home. So as long as the bottom of the proposed canyon is above home it will dive right in.

Reply:
This is not likely to be used for long distances if I understand it right. I'm working with a partner who has asked for something similar in plane RTL mode - ascend to a safe altitude and not got down into dips and valleys on the way back, but still go up over terrain that intervenes. Balancing AGL regulation limits is a challenge.

It does sound safer than what happens now.

Reply:
@tridge @timtuxworth @IamPete1  in such cased regulations are the least of our concerns. This is for safety and we not be ever used in normal cases. This is for flight failures. I am against dipping into valleys or canyons as it will be an almost guaranteed loss for radio/telemetry, which I think is more important than temporarily clipping any AGL regulations. Also much less risky staying up, as you cannot know how steep the next "wall" will be and if the aircraft can manage the accent rate while flying forward with 12-15m/s. Remember these are not going to stop then go vertical.  And @timtuxworth  long distance is relative, we want to be able to do this from up to 20km. That is up to the aircraft capability. I see no real down side.

Reply:
@tridge @IamPete1 Hi, I actually wanted to get some clarity on this aspect. What if the drone (for whatever reason) goes into QRTL at 900m AGL. Will it first descend to the altitude above terrain that is in the parameters? Say relative to the highest point in its path. Or will it not decent at all, only climb?
The later is not desirable and would be very dangerous. It should come down to at least the defined height over the highest terrain in its intended flight path home. If that is not possible then I would go with Peter and then request full terrain following, up and down.
The problem with staying high is say we are 10km out at 500m high and have 30% battery remaining when the QRTL kicks in, so it starts to fly home at 500m, half way home, the critical battery FS would kick in, which is Qland, and that would usually not be a problem from 50-100m AGL, however from 500m it will not have enough power to complete the decent. It would end up falling out of the sky half way down or so. Just as example.